nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,221 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.webhook;
+  defaultUser = "webhook";
+
+  hooksFile = pkgs.writeText "webhook-config" (
+    builtins.toJSON (
+      builtins.attrValues cfg.hooks
+    )
+  );
+
+  # https://github.com/adnanh/webhook/blob/master/docs/Hook-Rules.md
+  # We might merge with "or" (perhaps?), but until then, allow only a single definition.
+  typeTriggerRule = types.uniq types.attrs;
+
+in {
+  options = {
+    services.webhook = {
+      enable = mkEnableOption (mdDoc ''
+        [Webhook] is server written in Go, that allows you to create HTTP endpoints (hooks),
```

Comment:
Are you certain this is valid Markdown? I never knew about this syntax for links.

Reply:
Lgtm, but it's good to check. I think there's a formatting improvement to be made a bit further down ;)

Reply:
This is valid CommonMark called a shortcut reference link

https://spec.commonmark.org/0.30/#shortcut-reference-link

I don't know how to test whether https://search.nixos.org/options supports it, but if it uses CommonMark, it should. Is there documentation for that?

Reply:
You could build the manual locally: https://nixos.org/manual/nixos/unstable/#chap-contributing
I don't know if search.nixos.org has extra demands. @ncfavier?

Reply:
It's fine, we use options.json which only has rendered docbook.

Reply:
```suggestion
        [Webhook] is a server written in Go that allows you to create HTTP endpoints (hooks),
```

Reply:
It works locally for me. I also applied @ncfavier's suggestion.